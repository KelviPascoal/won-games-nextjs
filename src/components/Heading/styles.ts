import styled, { css, DefaultTheme } from 'styled-components';
import media from 'styled-media-query';
import { HeadingProps } from '.';

type WrapperProps = Omit<HeadingProps, 'children'>;
type LineColor = 'primary' | 'secondary';

const wrapperModifiers = {
  left: (theme: DefaultTheme, lineColor: LineColor) => css`
    padding-left: ${theme.spacings.xxsmall};
    border-left: 0.7rem solid ${!!lineColor && theme.colors[lineColor]};
  `,

  bottom: (theme: DefaultTheme, lineColor: LineColor) => css`
    position: relative;
    margin-bottom: ${theme.spacings.medium};
    &::after {
      position: absolute;
      left: 0;
      bottom: -1rem;
      content: '';
      width: 5rem;
      border-bottom: 0.4rem solid ${!!lineColor && theme.colors[lineColor]};
      background-color: ${theme.colors[lineColor]};
    }
  `
};

export const Wrapper = styled.h1<WrapperProps>`
  ${({ theme, lineDirection, textColor, lineColor }) => css`
    color: ${!!textColor && theme.colors[textColor]};
    font-size: ${theme.font.sizes.xlarge};
    ${media.greaterThan('medium')`
      font-size: ${theme.font.sizes.xxlarge};
    `}
    ${!!lineDirection &&
    !!lineColor &&
    wrapperModifiers[lineDirection](theme, lineColor)}
  `}
`;
