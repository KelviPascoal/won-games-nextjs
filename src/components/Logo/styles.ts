import styled, { css } from 'styled-components';

import { LogoProps } from '.';

export const Wrapper = styled.div<LogoProps>`
  ${({ theme, color }) => css`
    color: ${!!color && theme.colors[color]};
  `}
`;
