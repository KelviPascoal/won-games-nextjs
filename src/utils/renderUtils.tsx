import { GlobalStyles } from '../styles/global';
import { ThemeProvider } from 'styled-components';
import { ReactNode } from 'react';
import { theme } from '../styles/theme';
import { render as testRender } from '@testing-library/react';

export function render(children: ReactNode) {
  return testRender(
    <ThemeProvider theme={theme}>
      <GlobalStyles removeBg={true} />
      {children}
    </ThemeProvider>
  );
}
