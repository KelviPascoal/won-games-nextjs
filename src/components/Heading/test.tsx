import 'jest-styled-components';
import { screen } from '@testing-library/react';
import { theme } from 'styles/theme';
import { render } from 'utils/renderUtils';
import { Heading } from '.';

describe('<Heading />', () => {
  it('should render the heading', () => {
    render(<Heading>Won Games</Heading>);
    expect(
      screen.getByRole('heading', { name: /Won Games/i })
    ).toBeInTheDocument();
  });

  it('should render a white heading by default', () => {
    render(<Heading>Won Games</Heading>);

    expect(screen.getByRole('heading', { name: /Won Games/i })).toHaveStyle({
      color: theme.colors.white
    });
  });

  it('should render a black heading', () => {
    render(<Heading textColor="black">Won Games</Heading>);

    expect(screen.getByRole('heading', { name: /Won Games/i })).toHaveStyle({
      color: theme.colors.black
    });
  });

  it('should render a head with a primary color line on the left side', () => {
    render(<Heading lineDirection="left">Won Games</Heading>);

    expect(screen.getByRole('heading', { name: /Won Games/i })).toHaveStyle({
      'border-left': `0.7rem solid ${theme.colors.primary}`
    });
  });

  it('should render a head with a secondary color line on the left side', () => {
    render(
      <Heading lineDirection="left" lineColor="secondary">
        Won Games
      </Heading>
    );

    expect(screen.getByRole('heading', { name: /Won Games/i })).toHaveStyle({
      'border-left': `0.7rem solid ${theme.colors.secondary}`
    });
  });

  it('should render a head with a primary color line at the bottom side', () => {
    render(<Heading lineDirection="bottom">Won Games</Heading>);

    expect(screen.getByRole('heading', { name: /Won Games/i })).toHaveStyleRule(
      'border-bottom',
      `0.4rem solid ${theme.colors.primary}`,
      {
        modifier: '::after'
      }
    );
  });

  it('should render a head with a secondary color line at the bottom side', () => {
    render(
      <Heading lineDirection="bottom" lineColor="secondary">
        Won Games
      </Heading>
    );

    expect(screen.getByRole('heading', { name: /Won Games/i })).toHaveStyleRule(
      'border-bottom',
      `0.4rem solid ${theme.colors.secondary}`,
      {
        modifier: '::after'
      }
    );
  });

  it('should render a heading with a small size', () => {
    render(<Heading fontSize="small">Won Games</Heading>);
    expect(screen.getByRole('heading', { name: /won games/i })).toHaveStyle({
      'font-size': '1.6rem'
    });

    expect(screen.getByRole('heading', { name: /won games/i })).toHaveStyleRule(
      'width',
      '3rem',
      {
        modifier: '::after'
      }
    );
  });
});
