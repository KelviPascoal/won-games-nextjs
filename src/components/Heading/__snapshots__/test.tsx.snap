// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> should render a white heading by default 1`] = `
<h1
  class="sc-bczRLJ kltQOJ"
>
  Won Games
</h1>
`;

exports[`<Heading /> should render the heading 1`] = `
<h1
  class="sc-bczRLJ kltQOJ"
>
  test heading
</h1>
`;

exports[`<Heading /> should render the heading 2`] = `
<h1
  class="sc-bczRLJ kltQOJ"
>
  Won Games
</h1>
`;
